src/main/java/com/gizasystems/automationexercise/pages/ProductsPage.java

```diff
@@ -22,6 +24,15 @@ private By viewProduct_link(String productName) {
     private final By searchButton = By.xpath("//button[@id='submit_search']");
     private final By searchResult = By.xpath("//div[@class='productinfo text-center']//p");
 
+    private By hoverOnProduct(Integer Index) {
+        return By.xpath("(//div[@class=\"single-products\"]//img)[" + Index + "]");
+    }
+
+    private By ClickOnProduct(Integer Index) {
+        return By.xpath("(//a[@data-product-id=\"" + Index + "\"])[" + Index + "]");
+
+    }
```

Comment:
- Not sure to accept those locators as the user wouldn't know the index of the items if the order changes, this should be something like the name of the item, not the index
- Please follow the naming convention for the parameters variables (lowerCamaelCase).. Index should be index.